### PR TITLE
SWARM-1435: Introduce AsciidoctorJ fraction

### PR DESCRIPTION
**Pull request by gastaldi**
Motivation
----------
This includes the AsciidoctorJ fraction to the list of available fractions

Modifications
-------------
Added asciidoctorj fraction + tests

Result
------
AsciidoctorJ can now be executed inside a WildFly Swarm application without much fuzz.

### CODE REVIEW

**Comment by gastaldi** [comment]
Retest this please

File: pom.xml [hunk 1]
```diff
@@ -1381,6 +1387,8 @@
 
     <module>client-apis/health-api</module>
 
+    <module>fractions/asciidoctorj</module>
```
Comment (Ladicek) [inline pom.xml]: Should move to the `unsupported` profile? Same with the testsuite module below.

**Comment by Ladicek** [comment]
I'm just wondering about the value of this. Shouldn't I be able to do this just as part of the deployment without any issues? Or are there issues that this fraction eliminates?

File: pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply pom.xml]: +1 to move to unsupported

**Comment by kenfinnigan** [comment]
In theory yes, but there are manifest or module.xml dependency machinations required

**Comment by Ladicek** [comment]
If I'd have to touch module stuff when trying to deploy AsciidoctorJ, then a fraction is a perfect solution, thanks!

**Comment by gastaldi** [comment]
The Windows Build fails because of https://github.com/jruby/jruby/issues/3657. 

@bobmcwhirter any chance to have the JRuby team look at it?

**Comment by enebo** [comment]
@gastaldi we will try and reexamine our stale dll files issue.  The reason we have not fixed this so far is it is harder then it looks.  One engineer is really afraid of putting the dlls into a predictable location as hackers on Windows use that as a mechanism for replacing the dll.  Once we load the dll then windows is incapable of unloading it so we cannot delete the file from the JVM (although just saying that makes me wonder if there is an unload function).  Our binary launcher tries to overlay the JVM process so we do not necessarily have wrapper script which could attempt to clean the files up.

**Comment by gastaldi** [comment]
Apparently upgrading Asciidoctor helped on making the Windows tests pass. Let me ask to run the tests again, just in case

**Comment by gastaldi** [comment]
Retest this please

**Review by bobmcwhirter — APPROVED** [review]

**Pull request merged**